### PR TITLE
fix dump bypass_cache failure

### PR DESCRIPTION
**Pull request by kylazhang**
1. Change the fork place to make it more accurate for the child process executing
2. Add wait_for_login for test as it need more time to get the status during dump
3. Remove redundant kill and change the order for kill as it may cause error

### CODE REVIEW

**Comment by will-Do** [comment]
@kylazhang 
```
24.14s$ inspekt style
PEP8 disabled: E501,E265,W601,E402
/home/travis/build/autotest/tp-libvirt/libvirt/tests/src/virsh_cmd/domain/virsh_dump.py:244:1: E101 indentation contains mixed spaces and tabs
/home/travis/build/autotest/tp-libvirt/libvirt/tests/src/virsh_cmd/domain/virsh_dump.py:244:1: W191 indentation contains tabs
/home/travis/build/autotest/tp-libvirt/libvirt/tests/src/virsh_cmd/domain/virsh_dump.py:245:1: E101 indentation contains mixed spaces and tabs
/home/travis/build/autotest/tp-libvirt/libvirt/tests/src/virsh_cmd/domain/virsh_dump.py:247:1: E101 indentation contains mixed spaces and tabs
/home/travis/build/autotest/tp-libvirt/libvirt/tests/src/virsh_cmd/domain/virsh_dump.py:247:1: W191 indentation contains tabs
/home/travis/build/autotest/tp-libvirt/libvirt/tests/src/virsh_cmd/domain/virsh_dump.py:248:1: W191 indentation contains tabs
/home/travis/build/autotest/tp-libvirt/libvirt/tests/src/virsh_cmd/domain/virsh_dump.py:248:2: E101 indentation contains mixed spaces and tabs
/home/travis/build/autotest/tp-libvirt/libvirt/tests/src/virsh_cmd/domain/virsh_dump.py:249:1: W191 indentation contains tabs
/home/travis/build/autotest/tp-libvirt/libvirt/tests/src/virsh_cmd/domain/virsh_dump.py:249:2: E101 indentation contains mixed spaces and tabs
/home/travis/build/autotest/tp-libvirt/libvirt/tests/src/virsh_cmd/domain/virsh_dump.py:250:1: W191 indentation contains tabs
/home/travis/build/autotest/tp-libvirt/libvirt/tests/src/virsh_cmd/domain/virsh_dump.py:250:2: E101 indentation contains mixed spaces and tabs
/home/travis/build/autotest/tp-libvirt/libvirt/tests/src/virsh_cmd/domain/virsh_dump.py:251:1: W191 indentation contains tabs
/home/travis/build/autotest/tp-libvirt/libvirt/tests/src/virsh_cmd/domain/virsh_dump.py:251:2: E101 indentation contains mixed spaces and tabs
/home/travis/build/autotest/tp-libvirt/libvirt/tests/src/virsh_cmd/domain/virsh_dump.py:252:1: W191 indentation contains tabs
/home/travis/build/autotest/tp-libvirt/libvirt/tests/src/virsh_cmd/domain/virsh_dump.py:252:2: E101 indentation contains mixed spaces and tabs
/home/travis/build/autotest/tp-libvirt/libvirt/tests/src/virsh_cmd/domain/virsh_dump.py:253:1: W191 indentation contains tabs
/home/travis/build/autotest/tp-libvirt/libvirt/tests/src/virsh_cmd/domain/virsh_dump.py:253:2: E101 indentation contains mixed spaces and tabs
/home/travis/build/autotest/tp-libvirt/libvirt/tests/src/virsh_cmd/domain/virsh_dump.py:254:1: W191 indentation contains tabs
/home/travis/build/autotest/tp-libvirt/libvirt/tests/src/virsh_cmd/domain/virsh_dump.py:254:3: E101 indentation contains mixed spaces and tabs
/home/travis/build/autotest/tp-libvirt/libvirt/tests/src/virsh_cmd/domain/virsh_dump.py:256:1: E101 indentation contains mixed spaces and tabs
PEP8 check fail: /home/travis/build/autotest/tp-libvirt/libvirt/tests/src/virsh_cmd/domain/virsh_dump.py
PEP8 compliance FAIL

**Comment by kylazhang** [comment]
I will check ,thanks

File: libvirt/tests/src/virsh_cmd/domain/virsh_dump.py [hunk 1]
```diff
@@ -192,23 +192,6 @@ def check_dump_format(dump_image_format, dump_file):
...
-            child_pid = pid
-            try:
-                wait_pid_active(pid, timeout)
-            finally:
-                os.kill(child_pid, signal.SIGTERM)
-        else:
-            check_bypass(dump_file)
-            # Wait for parent process kills us
-            while True:
-                time.sleep(1)
```
Comment (chunfuwen) [inline libvirt/tests/src/virsh_cmd/domain/virsh_dump.py]: This could leads to dead loop since there is no skip condition here

Comment (kylazhang) [inline libvirt/tests/src/virsh_cmd/domain/virsh_dump.py]: hi @chunfuwen , it is child process and at last kill will stop it.

**Pull request merged**